src/functions.php

```diff
@@ -377,13 +386,132 @@ function upload(string $xml, $config)
         'PhonebookImportFile' => [
             'type' => 'text/xml',
             'filename' => 'updatepb.xml',
-            'content' => $xml,
+            'content' => $xmlNewPhonebook->asXML(), // convert XML object to XML string
         ]
     ];
 
-    $result = $fritz->postFile($formfields, $filefields); // send the command
-
+    $result = $fritz->postFile($formfields, $filefields); // send the command to store new phonebook
     if (strpos($result, 'Das Telefonbuch der FRITZ!Box wurde wiederhergestellt') === false) {
         throw new \Exception('Upload failed');
     }
 }
+
+
+/**
+ * Downloads the old phone book from Fritzbox
```

Comment:
s/Downloads the old phone book from Fritzbox/Download phone book from Fritzbox/

Reply:
OK